luna/gateware/interface/ulpi.py

```diff
@@ -1175,6 +1175,11 @@ def elaborate(self, platform):
             for address, value in platform.ulpi_extra_registers.items():
                 self.add_extra_register(address, value)
 
+        # Allow the platform to control ulpi clock handling; e.g. ULPI input clock operation on
+        # Spartan 6 plaforms needs double data rate buffer for the clock output from the link to the phy.
+        if hasattr(platform, 'ulpi_handle_clocking'):
```

Comment:
Can this be changed so the value of `handle_clocking` overrides this local attribute, if provided?

It seems cleaner to e.g. have `handle_clocking` default to `None`, and then have subsequent code replace `None` with `platform.ulpi_handle_clocking` or `True`; so an explicit value for `handle_clocking` in the instantiation wins.